cache.go

```diff
@@ -371,11 +361,14 @@ func (c *Cache[K, V]) Items() map[K]*Item[K, V] {
 }
 
 // Metrics returns the metrics of the cache.
-func (c *Cache[K, V]) Metrics() Metrics {
-	c.metricsMu.RLock()
-	defer c.metricsMu.RUnlock()
-
-	return c.metrics
+func (c *Cache[K, V]) Metrics() (m Metrics) {
+	// I imagine on most architectures this is equivalent to
+	// simply writing: return c.metrics
+	m.Insertions = atomic.LoadUint64(&c.metrics.Insertions)
```

Comment:
Maybe something like this would be more expressive, but it's just question of taste, ofc:
```
return Metrics{
	Insertions: atomic.LoadUint64(&c.metrics.Insertions),
	Hits:       atomic.LoadUint64(&c.metrics.Hits),
	Misses:     atomic.LoadUint64(&c.metrics.Misses),
	Evictions:  atomic.LoadUint64(&c.metrics.Evictions),
}
```

Reply:
That would be my suggestion as well.